test_runner/regress/test_proxy.py

```diff
@@ -140,26 +161,45 @@ def test_forward_params_to_client(static_proxy: NeonProxy, option_name: str):
         where name = any(%s)
     """
 
-    with static_proxy.connect(options=f"{option_name}=irrelevant") as conn:
+    with static_proxy.connect() as conn:
         with conn.cursor() as cur:
             cur.execute(query, (reported_params_subset,))
             for name, value in cur.fetchall():
                 # Check that proxy has forwarded this parameter.
                 assert conn.get_parameter_status(name) == value
 
 
-@pytest.mark.parametrize("option_name", ["project", "endpoint"])
 @pytest.mark.timeout(5)
-def test_close_on_connections_exit(static_proxy: NeonProxy, option_name: str):
+def test_close_on_connections_exit(static_proxy: NeonProxy):
     # Open two connections, send SIGTERM, then ensure that proxy doesn't exit
     # until after connections close.
-    with static_proxy.connect(options=f"{option_name}=irrelevant"), static_proxy.connect(
-        options=f"{option_name}=irrelevant"
-    ):
+    with static_proxy.connect(), static_proxy.connect():
         static_proxy.terminate()
         with pytest.raises(subprocess.TimeoutExpired):
             static_proxy.wait_for_exit(timeout=2)
         # Ensure we don't accept any more connections
         with pytest.raises(psycopg2.OperationalError):
-            static_proxy.connect(options=f"{option_name}=irrelevant")
+            static_proxy.connect()
     static_proxy.wait_for_exit()
+
+
+def test_sql_over_http(static_proxy: NeonProxy):
+    static_proxy.safe_psql("create role http with login password 'http'")
+
+    def q(sql: str, params: List[str] = []) -> Any:
+        connstr = f"postgresql://http:http@{static_proxy.domain}:{static_proxy.proxy_port}/postgres"
+        response = requests.post(
+            f"https://{static_proxy.domain}:{static_proxy.external_http_port}/sql",
+            data=json.dumps({"query": sql, "params": params}),
+            headers={"Content-Type": "application/sql", "X-Neon-ConnectionString": connstr},
+            verify= str(static_proxy.test_output_dir / "proxy.crt"),
+        )
+        assert response.status_code == 200
+        return response.json()
+
+    assert q('select 42 as answer') == [{'answer': 42}]
```

Comment:
what's about `update`/`delete` queries?

Reply:
Added tests for DDL and DML.

Unexpectedly, rust-postgres hadn't returned command tags, so any DML was quite blind. Changed driver one more time https://github.com/neondatabase/rust-postgres/pull/14/commits/7d3933c362c5142a0668bc205584485f17a61976

Seems that @skyzh stepped on the same thing already: https://github.com/sfackler/rust-postgres/pull/985/files (btw there is a bug -- "INSERT" unlike other tags returns two numbers where the first one is always zero and the second one is actual number of rows, https://www.postgresql.org/docs/current/protocol-message-formats.html#PROTOCOL-MESSAGE-FORMATS-COMMANDCOMPLETE)

In my case I needed command tag string too so did that in a way that is straightforward, but not compatible with the method that is currently in the latest version. A bit more work need to proper upstream that patch.

Reply:
what about DML queries with the `RETURNING` option?

Reply:
> what about DML queries with the RETURNING option?

no surprises, works as expected, added to tests